mypy/binder.py

```diff
@@ -125,7 +125,8 @@ def put(self, expr: Expression, typ: Type) -> None:
         if key not in self.declarations:
             assert isinstance(expr, BindableTypes)
             self.declarations[key] = get_declaration(expr)
-            self._add_dependencies(key)
+            if key is not None:
+                self._add_dependencies(key)
```

Comment:
I cannot comment there, but you can just add similar things for ``return self._get(expr.literal_hash)`` etc., or ``assert expr.literal_hash is not None`` where appropriate.